src/middleware.ts

```diff
@@ -0,0 +1,32 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+import { APIService } from "./services/api";
+
+export const config = {
+  matcher: "/panel/:path*",
+};
+
+const authPage = "/panel/auth";
+
+export default async function middleware(req: NextRequest) {
+  if (req.nextUrl.pathname === authPage) {
+    return NextResponse.next();
+  }
+  const isAuthenticated = await validateToken(req.cookies.getAll());
+  if (isAuthenticated) {
+    return NextResponse.next();
+  }
+  const redirectUrl = new URL(authPage, req.url);
+  return NextResponse.redirect(redirectUrl);
+}
+
+const validateToken = async (allCookies: any) => {
+  const response = await fetch(`${APIService.baseURL}/auth/validate`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
```

Comment:
This is a very common pattern used in requests. Maybe we could have a central service for generating these?

Reply:
I tried generalising the headers in a previous projects, it changes in every request and it becomes unnecessary, this also helps you see the headers and other info directly, so I believe it's better not to :+1: